hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAHSv2ClientImpl.java

```diff
@@ -107,6 +113,91 @@ public void testGetAppAttemptReport() throws IOException, YarnException {
         isEqualTo("test original tracking url");
   }
 
+  @Test
+  public void testGetContainerByAppAttempt() throws IOException, YarnException {
+    // Single container tests
+    final ApplicationId appId1 = ApplicationId.newInstance(0, 1);
+    final ApplicationId appId2 = ApplicationId.newInstance(0, 2);
+
+    final ApplicationAttemptId appAttemptId1 =
+            ApplicationAttemptId.newInstance(appId1, 1);
+    final ApplicationAttemptId appAttemptId2 =
+            ApplicationAttemptId.newInstance(appId2, 2);
+
+    final ContainerId containerId1 = ContainerId.newContainerId(appAttemptId1, 1);
+    final ContainerId containerId2 = ContainerId.newContainerId(appAttemptId2, 2);
+
+    // Multiple container test
+    final int multipleContainerSize = 4;
+    final int appMultipleId = 3;
+    final int appAttemptMultipleId = 3;
+    final ApplicationId appMultiple = ApplicationId.newInstance(0, appMultipleId);
+    final ApplicationAttemptId appAttemptIdMultiple =
+            ApplicationAttemptId.newInstance(appMultiple, appAttemptMultipleId);
+
+    List<TimelineEntity> containerEntities = new ArrayList<>();
+    for (int newEntityIdx = 0; newEntityIdx < multipleContainerSize; ++newEntityIdx) {
+      containerEntities.add(createContainerEntity(ContainerId.newContainerId(
+              appAttemptIdMultiple, containerId2.getContainerId() +1 + newEntityIdx)));
+    }
```

Comment:
To avoid offsetting the IDs (like the +3 below in assertThat(report.getContainerId().getContainerId()).isEqualTo(containerIdx + 3);) could you please move the multiple container case to another test method? Because currently if someone wants to add a new app+attempt+container he/she needs to edit the offset below which isn't trivial at first.